modules/wordads/php/widgets.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Widget for inserting an ad into your sidebar
+ *
+ * @since 0.1
+ * TODO: sanitize all the output
+ */
+class WordAds_Sidebar_Widget extends WP_Widget {
+
+	private static $allowed_tags = array( 'mrec', 'wideskyscraper' );
+
+	private $options = array();
+
+	function __construct() {
+		parent::__construct(
+			'wordads_sidebar_widget',
+			__( 'WordAds Widget (Jetpack)', 'jetpack' ),
+			array( 'description' => __( 'Insert a WordAds ad wherever you can place a widget.', 'jetpack' ) )
```

Comment:
It might be good to add [selective refresh support](https://make.wordpress.org/core/2016/02/16/selective-refresh-in-the-customizer/) to the widget, since we tend to do it for all new widgets these days.